### DISCUSSION TITLE
signinUrl and callbackUrl not correct after deployment

### DISCUSSION THREAD

**Discussion by hyunyikim** [nextauthjs/next-auth#9430] [Help]
Hi, I tried to implement google oauth login, but I've encountered issues with the callback url and signin url being wrong. 
I've been struggling with this problem for a week now. If anyone has any insights or clues on how to resolve this, I would greatly appreciate your help!

My project have two environment variables. 
- development
- production

I deployed the project using development environment variables on AWS Amplify.
When testing google oauth login, I noticed that callback url and signin url were pointing to the production url instead of development url. I checked that the project is reading the correct environment variables, and everything seems to be in order. 
But when I run the same code locally, not deployed, both urls point to the development url as I expected.

I initialloy thought that `NEXTAUTH_URL` would serve as the callback url and signin url, but I'm puzzled as to why it's not functioning as expected. Any guidance on resolving this issue would be highly valued.

### code 
```js
// package.json
{
    "scripts": {
        "dev": "next dev",
        "build:dev": "env-cmd -f .env.development next build",
        "build:prod": "env-cmd -f .env.production next build",
        "start": "next start",
        "lint": "next lint"
    },
    "dependencies": {
        "next": "^13.2.3",
        "next-auth": "^4.23.1",
        "react": "^18.2.0",
        "react-dom": "^18.2.0",
    },
}
```
```
//.env.development
GOOGLE_ID=AAAA
GOOGLE_SECRET=BBB
SECRET=CCC
NEXTAUTH_SECRET=DDD
NEXTAUTH_URL=https://dev-console.hello.co.kr

//.env.production
GOOGLE_ID=AAAA
GOOGLE_SECRET=BBB
SECRET=CCC
NEXTAUTH_SECRET=DDD
NEXTAUTH_URL=https://console.hello.co.kr
```
```js
// [...nextauth].js
import NextAuth from 'next-auth';
import GoogleProvider from 'next-auth/providers/google';

export const authOptions = {
	providers: [
		GoogleProvider({
			clientId: process.env.GOOGLE_ID,
			clientSecret: process.env.GOOGLE_SECRET,
		}),
	],
	secret: process.env.NEXTAUTH_SECRET,
};

export default NextAuth(authOptions);
```
```js
// LoginButton
<button
	onClick={() => {
		signIn('google', {
			callbackUrl: '/dashboard',
		});
	}}
	style={{
		background: 'none',
		border: 'none',
		cursor: 'pointer',
	}}>
	<Image
		src={'/assets/images/auth/sign-in-with-google.png'}
		alt="sign-in-google"
		width={240}
		height={50}
	/>
</button>
```

### console.log in browser
```js
useEffect(() => {
(async () => {
	const proivders = await getProviders();
	console.log('providers :>> ', proivders);
        })();
}, []);
```
<img width="691" alt="스크린샷 2023-12-21 오후 7 15 28" src="https://github.com/nextauthjs/next-auth/assets/38061822/04fe4043-a0bf-4d18-89fd-de774f534174">

**Comment by Di0nza** [comment]
same problem, did u solve it ? I will be very grateful if you tell me
